ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt

```diff
@@ -115,10 +127,43 @@ private fun SelectPackageButton(
         shape = RoundedCornerShape(15.dp),
         border = border,
     ) {
-        Text(
-            text = "Purchase ${packageInfo.rcPackage.identifier}. " +
-                "Price: ${packageInfo.rcPackage.product.price.formatted}",
-        )
+        Column(
+            modifier = Modifier.fillMaxSize().padding(4.dp),
+            verticalArrangement = Arrangement.spacedBy(4.dp),
+        ) {
+            Row(horizontalArrangement = Arrangement.spacedBy(6.dp), verticalAlignment = Alignment.CenterVertically) {
+                CheckmarkBox(isSelected = isSelected, colors = state.currentColors)
+                Text(
+                    text = packageInfo.localization.offerName ?: packageInfo.rcPackage.product.title,
+                )
+            }
+            // TODO-PAYWALLS: Substitute with intro eligibility composable
+            val pricingPhases = packageInfo.rcPackage.product.defaultOption?.pricingPhases
+            val offerText = if (pricingPhases != null && pricingPhases.size > 1) {
+                packageInfo.localization.offerDetailsWithIntroOffer
+            } else {
+                packageInfo.localization.offerDetails
+            }
+            Text(text = offerText ?: "")
+        }
+    }
+}
+
+@Composable
+private fun CheckmarkBox(isSelected: Boolean, colors: TemplateConfiguration.Colors) {
+    Box(
+        modifier = Modifier
+            .size(Template2UIConstants.checkmarkSize)
+            .clip(CircleShape)
+            .background(colors.accent2.copy(alpha = Template2UIConstants.checkmarkUnselectedAlpha)),
+    ) {
+        if (isSelected) {
+            Icon(
+                painter = painterResource(id = R.drawable.check_circle),
+                contentDescription = null,
+                modifier = Modifier.fillMaxSize(),
+            )
```

Comment:
I wonder if this could use `PaywallIIcon` instead?

Reply:
We can indeed use it. However, `PaywallIcon` requires a tint color so we will need to pass it in instead of relying on the button automatically tinting it up for us... (Icon is automatically tinted by the surrounding button if you don't pass the tint property). Still, this is probably good so I will change it 👍

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1301/commits/3f95a412d7c9e861b4d785eab43731617ac77f22